src/parser/csql_grammar.y

```diff
@@ -3645,6 +3729,83 @@ alter_stmt
 			$$ = node;
 			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
 
+		DBG_PRINT}}
+	| ALTER			/* 1 */
+		{ push_msg (MSGCAT_SYNTAX_INVALID_ALTER_SYNONYM); }	/* 2 */
+	  opt_access_modifier	/* 3 */
+	  SYNONYM		/* 4 */
+	  simple_path_id	/* 5 */
+	  For			/* 6 */
+	  simple_path_id       	/* 7 */
+	  opt_comment_spec	/* 8 */
+		{{
+
+			PT_NODE *p = parser_new_node(this_parser, PT_ALTER_SYNONYM);
+
+			if (p)
+			  {
+			    p->info.alter_synonym.access_modifier = $3;
+
+			    /* synonym_name, synonym_owner_name */
+			    PT_NODE *synonym = $5;
+			    if (synonym && synonym->node_type == PT_DOT_)
+			      {
+				assert (synonym->info.dot.arg1 && synonym->info.dot.arg2->node_type == PT_NAME);
```

Comment:
assert (synonym->info.dot.arg1 && synonym->info.dot.arg1->node_type == PT_NAME);

Reply:
I found more code related to this typo. I'll fix it together.